### PR TITLE
Terrain estimator

### PR DESCRIPTION
**Pull request by tumbili**

### CODE REVIEW

File: makefiles/nuttx/config_px4fmu-v2_default.mk [hunk 1]
```diff
@@ -121,6 +121,7 @@ MODULES		+= lib/geo
 MODULES		+= lib/geo_lookup
 MODULES		+= lib/conversion
 MODULES		+= lib/launchdetection
+MODULES 	+= lib/terrain_estimation
```
Comment (LorenzMeier) [inline makefiles/nuttx/config_px4fmu-v2_default.mk]: Code style

**Comment by LorenzMeier** [comment]
@tumbili Travis file again. Please register so you get the feedback yourself and can fix in realtime.

**Comment by LorenzMeier** [comment]
Still no luck with Travis. The build log should have the details.

**Comment by tumbili** [comment]
Should work now :-/

**Comment by tumbili** [comment]
He doesn't like something with the Vectors, I'll fix this tomorrow, need to sleep.

**Comment by tumbili** [comment]
This is the error I get
```
/home/roman/Firmware//src/lib/eigen/Eigen/src/Core/PlainObjectBase.h:857:7: fatal error: static_assert failed "INVALID_MATRIX_TEMPLATE_PARAMETERS"
      EIGEN_STATIC_ASSERT((EIGEN_IMPLIES(MaxRowsAtCompileTime==1 && MaxColsAtCompileTime!=1, (Options&RowMajor)==RowMajor)
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/roman/Firmware//src/lib/eigen/Eigen/src/Core/util/StaticAssert.h:32:40: note: expanded from macro 'EIGEN_STATIC_ASSERT'
    #define EIGEN_STATIC_ASSERT(X,MSG) static_assert(X,#MSG);
                                       ^             ~
/home/roman/Firmware//src/lib/eigen/Eigen/src/Core/Map.h:126:30: note: in instantiation of member function 'Eigen::PlainObjectBase<Eigen::Matrix<float, 3, 1, 1, 3, 1> >::_check_template_params' requested here
      PlainObjectType::Base::_check_template_params();
                             ^
/home/roman/Firmware/src/lib/mathlib/math/Matrix.hpp:311:52: note: in instantiation of member function 'Eigen::Map<Eigen::Matrix<float, 3, 1, 1, 3, 1>, 0, Eigen::Stride<0, 0> >::Map' requested here
                Eigen::Matrix<float, M, N, Eigen::RowMajor> Me = Eigen::Map<Eigen::Matrix<float, M, N, Eigen::RowMajor> >
                                                                 ^
/home/roman/Firmware/src/lib/terrain_estimation/terrain_estimator.cpp:100:15: note: in instantiation of function template specialization 'math::MatrixBase<3, 1>::operator*<3>' requested here
               B * R * B.transposed() + Q) * dt;
                     ^
1 error generated.
make[3]: *** [terrain_estimator.cpp.o] Error 1
make[2]: *** [/home/roman/Firmware/Build/posix_sitl.build//home/roman/Firmware/src/lib/terrain_estimation/module.pre.o] Error 2
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [/home/roman/Firmware/Build/posix_sitl.build/firmware.a] Error 2
make: *** [posix] Error 2

```

**Comment by tumbili** [comment]
@TSC21 Any chance of having a look what's happening here?

**Comment by tumbili** [comment]
@AndreasAntener @SimonWilks Did you already do some tests with this?

**Comment by tumbili** [comment]
@AndreasAntener @SimonWilks I just corrected a small mistake in the estimator, shouldn't make much difference though. BTW, I'll be back today in the evening, so looking forward to some flight tests :-)

**Comment by TSC21** [comment]
@tumbili check the assertion. Probably there is some error on the matrix construction.

**Comment by AndreasAntener** [comment]
@tumbili No, coming up this week I hope if the weather plays along :)

**Comment by mhkabir** [comment]
@tumbili Would you quickly mind doing the GPS as a velocity estimate from the main filter instead, like we discussed? Thanks! :)

**Comment by AndreasAntener** [comment]
Open TODOs: #2964

**Comment by AndreasAntener** [comment]
Closing in favor of #2969

**Pull request closed without merging**